docs/reports/tootsie-retro.md

```diff
@@ -0,0 +1,416 @@
+for buying IBM, etc.
+
+We used Levanter's implementation of the Llama architecture.
+We trained with a sequence length of 4096 tokens per sample.
+We used JAX's TPU Splash Attention kernel.
+
+We used mixed float32/bfloat16 precision, with parameters and optimizer states in float32 and compute in bfloat16
+(except for the final softmax over the vocabulary.)
+We used the AdamW optimizer.
+
+### Tokenizer
+
+In Marin, we also standardized on the Llama 3 tokenizer, after [an experiment](https://github.com/marin-community/marin/issues/524) showing it to be superior
+to both Llama 2 and NeoX in terms of bits-per-byte (bpb).
+
+### Batch Schedule
+
+We used a varying batch schedule, with between 1024 * 4096 = 4Mi tokens, 3072 * 4096 = 12 Mi tokens and 4096 * 4096 = 16 Mi tokens.
+(As with most things Tootsie, we did not initially plan for this schedule.)
+(Note, we use `Mi` to mean Mebi-, not mega-.)
+
+### Checkpointing Policy
+
+We saved permanent full checkpoints every 20,000 steps (which, due to the varying batch schedule, could be a varying number of tokens.)
+To save on egress bandwidth charges, we have not made these available. However, if they are of interest we should be
+able to share them.
+
+("Temporary" checkpoints were saved much more frequently.)
+
+# Training Phases
+
+Retrospectively, we can partition the 8b run into several different phases. We have given them animal names as monikers.
+
+* *DCLM WSD-S Phase*: In the first phase, we used the "DCLM mix" and [WSD-S](https://arxiv.org/abs/2410.05192)  for about 2.8T tokens. We used 2x TPU v5e-256 coordinated with multislice for this.
+* *DCLM EMA Phase*: We were given access to a v4-2048 slice and moved to that. (So fast!) To better utilize the hardware, we increased our batch size 50%. We also switched from WSD-S to WSD, using the exponential moving averaging (EMA) of weights for monitoring evaluation performance. (XXX cite whoever) We kept the learning rate high until 3.7T tokens.
+* *First Cooldown: Jellyfish*  It was time to cooldown, as we were starting to run low on DCLM. Following recent work on midtraining (XXX), we decided to fold in higher quality data. Ultimately, after a series of ablations (XXX link) we landed on roughly the Dolmino mixture, with a few tweaks. The checkpoint at the end of this cooldown we call "monumental-jellyfish", and it had some fairly solid results. XXX
+* *Reheated: Phoenix* We had more time for training, so we decided to keep going. We rapidly rewarmed the model and transitioned our mixture to [Nemotron-CC ](XXX cite)XXX(plus some code and math).
+* *Second Cooldown: Starling*: Now we were running low on time, so we started another cooldown. We followed a similar process to the first cooldown, but added a few new datasets that we had created and also some that had dropped since our previous attempt. We call the result of this cooldown "starling."
+
+We emphasize that these phases are retrospective: we didn't know exactly what changes we would make.
```

Comment:
Why are these retrospective...I don't think these names are dependent on the future?